File: 2300-successful-pairs-of-spells-and-potions/2300-successful-pairs-of-spells-and-potions.py
class Solution:
    def successfulPairs(self, spells: List[int], potions: List[int], success: int) -> List[int]:
        
        n = len(spells)
        m = len(potions)
        
        answer = list()
        potions.sort()
        for i in range(n):
            
            left, right = 0, m - 1
            
            while left <= right:
                
                mid = (left + right) // 2
                
                if potions[mid] * spells[i] >= success:
                    right = mid - 1
                
                else:
                    left = mid + 1
                    
            answer.append(m - left)
            
        return answer

File: 2300-successful-pairs-of-spells-and-potions/2300-successful-pairs-of-spells-and-potions.js
/**
 * @param {number[]} spells
 * @param {number[]} potions
 * @param {number} success
 * @return {number[]}
 */
var successfulPairs = function(spells, potions, success) {
    
    let n = spells.length
    let m = potions.length
    
    let answer = new Array(n)
    
    potions.sort((a,b)=>{
        if (a<b){
            return -1
        }
        else if (a>b){
            return 1
        }
        else{
            return 0
        }
    })
    
    for (let i = 0; i < n; i++){
        
        let left = 0
        let right = m - 1
        
        while (left <= right){
            
            let mid = Math.floor((left + right) / 2)
            
            if (potions[mid] * spells[i] >= success){
                right = mid - 1
            }
            else {
                left = mid + 1
            }
        }
        
        answer[i] = m - left
    }
    
    return answer
};